modules/openapi-generator/src/main/resources/typescript-fetch/modelOneOf.mustache

```diff
@@ -31,7 +31,7 @@ export function {{classname}}FromJSONTyped(json: any, ignoreDiscriminator: boole
     }
 {{/discriminator}}
 {{^discriminator}}
-    return { {{#oneOf}}...{{{.}}}FromJSONTyped(json, true){{^-last}}, {{/-last}}{{/oneOf}} };
+    return {{#oneOf}}{{{.}}}FromJSONTyped(json, true){{^-last}} || {{/-last}}{{/oneOf}};
```

Comment:
this works for enums - but breaks the object models (the original code made more senses for objects).
perhaps need some condition here to process enums and objects differently.  might need to create a flag for that in the java model, if it's not available yet

Reply:
I think my lack of understanding is preventing us from moving forward so please bare with me.

I added an example of a `oneOf` using two objects

https://github.com/OpenAPITools/openapi-generator/pull/10017/files#diff-5a7c250c7d123900a98f6cb955465d37c8a93c4744a5d7e3798340fe76dbbd63

Why was is template not used to render the object?

I think it used https://github.com/OpenAPITools/openapi-generator/blob/bf89f40263c6a57a76697c551d2c30c651812f7a/modules/openapi-generator/src/main/resources/typescript-fetch/modelGeneric.mustache#L78 instead

Reply:
great question...

Reply:
Ah, right. It's because your `ObjectCombinedOneOf` model is not `oneOf`. It's a regular object model with a single property.
This property, in turns, is defined as `oneOf` - but since it's inlined, there is no a separate top-level definition for that.
Probably try something like:
```
ObjectCombinedOneOf:
    oneOf:
        - $ref: '#/components/schemas/PatternObject'
        - $ref: '#/components/schemas/PatternObjectTwo'          
```

Reply:
```suggestion
    return {{#oneOf}}{{{.}}}FromJSONTyped(json, true){{^-last}} ?? {{/-last}}{{/oneOf}};
```